catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostProcessPlugin.java

```diff
@@ -0,0 +1,568 @@
+  }
+
+  public NitfPostProcessPlugin(Supplier<NitfRenderer> nitfRendererSupplier) {
+    this.nitfRendererSupplier = nitfRendererSupplier;
+  }
+
+  @Override
+  public ProcessRequest<ProcessCreateItem> processCreate(ProcessRequest<ProcessCreateItem> input) {
+    Validate.notNull(input, "processCreate(): argument 'input' may not be null.");
+
+    handleProcessCreateItem(input.getProcessItems());
+    return input;
+  }
+
+  @Override
+  public ProcessRequest<ProcessUpdateItem> processUpdate(ProcessRequest<ProcessUpdateItem> input) {
+    Validate.notNull(input, "processUpdate(): argument 'input' may not be null.");
+
+    handleProcessUpdateItem(input.getProcessItems());
+    return input;
+  }
+
+  @Override
+  public ProcessRequest<ProcessDeleteItem> processDelete(ProcessRequest<ProcessDeleteItem> input) {
+    Validate.notNull(input, "processDelete(): argument 'input' may not be null.");
+
+    return input;
+  }
+
+  private boolean isNitfMimeType(String rawMimeType) {
+    try {
+      return NITF_MIME_TYPE.match(rawMimeType);
+    } catch (MimeTypeParseException e) {
+      LOGGER.debug("unable to compare mime types: {} vs {}", NITF_MIME_TYPE, rawMimeType);
+    }
+
+    return false;
+  }
+
+  public void setMaxSideLength(int maxSideLength) {
+    if (maxSideLength > 0) {
+      LOGGER.trace("Setting derived image maxSideLength to {}", maxSideLength);
+      this.maxSideLength = maxSideLength;
+    } else {
+      LOGGER.debug(
+          "Invalid `maxSideLength` value [{}], must be greater than zero. Default value [{}] will be used instead.",
+          maxSideLength,
+          DEFAULT_MAX_SIDE_LENGTH);
+      this.maxSideLength = DEFAULT_MAX_SIDE_LENGTH;
+    }
+  }
+
+  public void setMaxNitfSizeMB(int maxNitfSizeMB) {
+    this.maxNitfSizeMB = maxNitfSizeMB;
+  }
+
+  public void setCreateOverview(boolean createOverview) {
+    this.createOverview = createOverview;
+  }
+
+  public void setStoreOriginalImage(boolean storeOriginalImage) {
+    this.storeOriginalImage = storeOriginalImage;
+  }
+
+  private void handleProcessCreateItem(List<ProcessCreateItem> processCreateItems) {
+    List<ProcessCreateItem> createdItems = new ArrayList<>();
+    processCreateItems.stream().map(this::handleProcessCreateItem).forEach(createdItems::addAll);
+    processCreateItems.addAll(createdItems);
+  }
+
+  private List<ProcessCreateItem> handleProcessCreateItem(ProcessCreateItem processCreateItem) {
+    List<ProcessCreateItem> createdItems = new ArrayList<>();
+    Metacard metacard = processCreateItem.getMetacard();
+    ProcessResource processResource = processCreateItem.getProcessResource();
+
+    String mimeType = processResource.getMimeType();
+
+    if (isNitfMimeType(mimeType)) {
+      if (processResource.getSize() / MEGABYTE <= maxNitfSizeMB) {
+        try (TemporaryFileBackedOutputStream fbos =
+            new TemporaryFileBackedOutputStream(DEFAULT_FILE_BACKED_OUTPUT_STREAM_THRESHOLD)) {
```

Comment:
❓ How does `DEFAULT_FILE_BACKED_OUTPUT_STREAM_THRESHOLD` relate to `maxNitfSizeMB`?